core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala

```diff
@@ -787,6 +788,31 @@ class DynamicBrokerConfigTest {
     verifyIncorrectLogLocalRetentionProps(2000L, 1000L, -1, 100)
   }
 
+  @Test
+  def testUpdateDynamicRemoteLogManagerConfig(): Unit = {
```

Comment:
Could we also write a test that asserts that the index files on disk (at $logdir/remote-log-index-cache) belonging to older cache gets deleted after resize if done? Otherwise, we may end up with a file leak after resize.

You can create this test at RemoteIndexCacheTest.

Reply:
There was indeed files leak and I fixed it.